src/egon/data/datasets/hydrogen_etrago/h2_grid.py

```diff
@@ -0,0 +1,128 @@
+"""The central module containing all code dealing with heat sector in etrago
+"""
+from geoalchemy2.types import Geometry
+import geopandas as gpd
+
+from egon.data import db
+from egon.data.datasets.etrago_setup import link_geom_from_buses
+from egon.data.datasets.gas_prod import assign_bus_id
+from egon.data.datasets.scenario_parameters import get_sector_parameters
+
+
+def insert_h2_pipelines():
+    """ Insert hydrogen grid to etrago table based on CH4 grid."""
+    H2_buses = db.select_geodataframe(
+        f"""
+        SELECT * FROM grid.egon_etrago_bus WHERE scn_name = 'eGon100RE' AND
+        carrier IN ('H2_grid', 'H2_saltcavern') and country = 'DE'
+        """,
+        epsg=4326,
+    )
+
+    pipelines = db.select_geodataframe(
+        f"""
+        SELECT * FROM grid.egon_etrago_link
+        WHERE scn_name = 'eGon2035' AND carrier = 'CH4'
+        AND bus0 IN (
+            SELECT bus_id FROM grid.egon_etrago_bus
+            WHERE scn_name = 'eGon2035' AND country = 'DE'
+        ) AND bus1 IN (
+            SELECT bus_id FROM grid.egon_etrago_bus
+            WHERE scn_name = 'eGon2035' AND country = 'DE'
+        );
+        """
+    )
+
+    CH4_H2_busmap = db.select_dataframe(
+        f"""
+        SELECT * FROM grid.egon_etrago_ch4_h2 WHERE scn_name = 'eGon2035'
+        """,
+        index_col="bus_CH4",
+    )
+
+    scn_params = get_sector_parameters("gas", "eGon100RE")
+
+    pipelines["carrier"] = "H2_retrofit"
+    pipelines["p_nom"] *= 0.6  # capacity retrofitting factor
```

Comment:
Should be fixed by p-e-s run, will be fixed in https://github.com/openego/eGon-data/issues/686 I guess.